tests/src/unit-bjdata.cpp

```diff
@@ -1514,11 +1542,8 @@ TEST_CASE("BJData")
                     // create expected byte vector
                     std::vector<std::uint8_t> expected;
                     expected.push_back(static_cast<std::uint8_t>('['));
-                    if (N != 0)
```

Comment:
Seeing existing test cases change - is this a breaking change for existing clients?

Reply:
Yes - this is a breaking change for encoding, but fixes roundtrip encoding/decoding.

Before:
- `json::binary` gets encoded as a typed array of `U`
- typed array of `U` gets decoded as `json::array` ❗does not match encoded data

After:
- `json::binary` gets encoded as a typed array of `B`
- typed array of `B` gets decoded as `json::binary` ✅ matches encoded data

I could introduce a new flag to continue encoding binary as `uint8` if the breaking change is undesirable?

Reply:
We must not break existing code, so yes, a parameter would make sense.